src/main/java/com/chrosciu/bootcamp/tasks/github/GithubClient.java

```diff
@@ -10,22 +10,27 @@
     private final GithubApi githubApi;
 
     public Flux<Repository> getUserRepositories(String username) {
-        //TODO: Implement
-        return null;
+        Mono<List<Repository>> result = githubApi.getUserRepositories(username);
+        return result.flatMapMany(Flux::fromIterable);
     }
 
     public Flux<Branch> getUserRepositoryBranches(String username, String repo) {
-        //TODO: Implement
-        return null;
+        Mono<List<Branch>> result = githubApi.getUserRepositoryBranches(username, repo);
+        return result.flatMapMany(Flux::fromIterable);
     }
 
     public Flux<Repository> getUsersRepositories(Flux<String> usernames) {
-        //TODO: Implement
-        return null;
-    }
+        return usernames
+                .flatMap(username
+                        -> getUserRepositories(username));
+    };
 
     public Flux<String> getAllUserBranchesNames(String username) {
-        //TODO: Implement
-        return null;
+        return getUserRepositories(username)
+                .flatMap(repo
+                        -> getUserRepositoryBranches(username, repo.getName()))
+                .distinct()
```

Comment:
To zadziała trochę z przypadku bo klasa Branch ma tylko jedno pole name tak więc faktycznie dwa obiekty typu Branch o tej samej wartości tego pola będą sobie równe - co jakby jednak tych pól było więcej ? Czy na pewno ten operator wpięty jest w dobre miejsce ?